### ISSUE TITLE
Can't retain control over BluetoothDevice when retrieved again

### ISSUE THREAD

**Issue by froqstar** [robolectric/robolectric#6814]
### Description

I want to test whether my code correctly retrieves the bonding state of a `BluetoothDevice`. 
The class under test retrieves the respective device via its address by invoking the `BluetoothAdapter.getRemoteDevice(address)` function.

The problem is that robolectric seems to create a new `BluetoothDevice` instance for every invocation of `getRemoteDevice()`.
Thus, if I obtained a `Shadow` of the device to set the bond state, this has no effect on the instance retrieved by the class under test.

### Steps to Reproduce
```kotlin
@Test
fun `should retain control across invocations`() {
    val address = "12:34:56:78:90:00"
    val givenBondState = BluetoothDevice.BOND_BONDING

    val originalDevice = getBluetoothDevice(address)
    val deviceShadow = shadowOf(originalDevice)
    deviceShadow.setBondState(givenBondState)

    val secondDevice = getBluetoothDevice(address)

    assertThat(originalDevice.bondState).isEqualTo(givenBondState) // success
    assertThat(secondDevice.bondState).isEqualTo(givenBondState) // fails
    assertThat(secondDevice).isSameAs(originalDevice) // also fails
}
private fun getBluetoothDevice(address: String): BluetoothDevice =
    (ApplicationProvider.getApplicationContext<Context>()
        .getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager).adapter.getRemoteDevice(address)
```

### Robolectric & Android Version
robolectric: 4.6.1
SDK: 28 (doesn't matter if specified or not)

**Comment by utzcoz** [comment]
Looks like `BluetoothAdapter` will create a new `BluetoothDevice` instance when you call `getRemoteDevice` every time(https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/bluetooth/BluetoothAdapter.java;l=854-864?q=getRemoteDevice). Maybe you should cache `BluetoothDevice` with address manually for your need.

**Comment by hoisie** [comment]
Yeah it looks like getting a different BluetoothDevice object is WAI. 

However, BluetoothDevice.getBondState should be shadowed and query the state in ShadowBluetoothAdapter?

**Comment by hoisie** [comment]
ok looks like BluetoothDevice.getBondState is already [shadowed](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java#L179), but it does not share state with ShadowBluetoothAdapter.

**Comment by utzcoz** [comment]
> ok looks like BluetoothDevice.getBondState is already [shadowed](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowBluetoothDevice.java#L179), but it does not share state with ShadowBluetoothAdapter.

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/bluetooth/BluetoothDevice.java;l=1603-1616;drc=master Looks like the `IBluetooth` service should cache bond state for `BluetoothDevice` with its `mAttributeSource`, shared from `BluetoothAdpater`(https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/bluetooth/BluetoothAdapter.java;l=862;drc=master?q=getRemoteDevice).
